lib/core/src/server/common/custom-presets.js

```diff
@@ -3,5 +3,10 @@ import serverRequire from '../utils/server-require';
 
 export default function loadCustomPresets({ configDir }) {
   const presets = serverRequire(path.resolve(configDir, 'presets'));
-  return presets || [];
+  const main = serverRequire(path.resolve(configDir, 'main'));
```

Comment:
Can we do the same sanity check here, to make `main` / `presets` mutually exclusive?